notebooks/NNFullCombos.py

```diff
@@ -0,0 +1,446 @@
+    qdp = np.linspace(qdp_min, qdp_max, array_length, endpoint=True)  # Flow at the current segment in the previous timestep
+    qlat = np.linspace(qlat_min, qlat_max, array_length, endpoint=True) # lat inflow into current segment in the current timestep
+    velp = .5  # Velocity in the current segment in the previous timestep NOT USED AS AN INPUT!!!
+    depthp = np.linspace(depthp_min ,depthp_max , array_length, endpoint=True) # D
+
+
+    
+    #this is a normalization function that is used to scale the input variables between 0-1 to make the network train more efficiently. If not normalized some values may influence the NN too much
+    def normalize(val,max,min,target_max=1,target_min=0):
+        return (val-min)/(max-min)*(target_max-target_min)+target_min
+
+
+#MC function that allows us to calculate the expected outputs used as our Y output. Can be used to generate real calculations from the MC code.
+    def singlesegment(
+            dt # dt
+            , qup = None # qup can be anything
+            , quc = None # quc will not be more than a 10 percent diff than qup
+            , qlat = None # ql can be anything - key 
+            , qdp = None # qdp will not be more than 20 percent diff than qup+qlat
+            , dx = None # dx fully variable 
+            , bw = None # bw correlated to tw, tw always > bw
+            , tw = None # tw correlated to bw, bw always < tw
+            , twcc = None # twcc always > than tw, tw of broader floodplain
+            , n_manning = None # usually smaller than n_manning_cc
+            , n_manning_cc = None # ncc usually greater than n_manning
+            , cs = None # cs correlated to bw and tw
+            , s0 = None # s0 variable 
+            , velp = None # velocity at previous time step not rel
+            , depthp = None # depth at previous time step starting point for iteration depthp = approx(y_direct(bw,n_manning,s0,avg(qup,qdp)))
+        ):
+
+        
+        
+
+        
+        # call Fortran routine
+        return mc.muskingcungenwm(
+            dt, qup, quc, qdp, qlat, dx, bw, tw, twcc
+            ,n_manning, n_manning_cc, cs, s0, velp, depthp
+        )
+        #return qdc, vel, depth
+        
+        
+#for loops used to find every combination of our variables possible based on our original array size between the mins and maxs of each variable.
+    Y = []
+    M = []
+
+    for qup_i in range(0,(array_length)-1):
+        for quc_j in range(0,(array_length)-1):
+            for qlat_k in range(0,(array_length)-1):
+                for qdp_l in range(0,(array_length)-1):
+                    for bw_n in range(0,(array_length)-1):
+                        for tw_o in range(0,(array_length)-1):
+                            for cs_p in range(0,(array_length)-1):
+                                for s0_q in range(0,(array_length)-1):
+                                    for depthp_s in range(0,(array_length)-1):
```

Comment:
for loop mania!!!  Is it possible to refactor this using something like https://docs.python.org/3/library/itertools.html#itertools.combinations?
Might be much more readable, and should be able to grab a generator of the results to normalize so should be efficient.